pkg/auth/jwt_auth_cache.go

```diff
@@ -0,0 +1,78 @@
+package auth
+
+import (
+	"bytes"
+	"context"
+	"io/ioutil"
+	"time"
+
+	"github.com/dgrijalva/jwt-go"
+	"github.com/grpc-ecosystem/go-grpc-middleware/auth"
+
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+)
+
+type jwtAuthCache struct {
+	headerScheme string
+	jwtSecret    []byte
+	tokenCache   map[string]time.Time
```

Comment:
There is no guarantee that this cache is used synchronously, right? I guess you'll need to place a mutex around `tokenCache` and `cacheOrder` accesses.

Reply:
Yes, that was one of my concern and given that I don't know much about the Go threading model, I wasn't too sure about what the best approach would be. So let's protect the cache with a mutex then.